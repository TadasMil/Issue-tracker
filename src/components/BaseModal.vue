<script setup lang="ts" name="BaseModal">
const { title, close } = defineProps<{
  title: string
  close: () => void
}>()
</script>

<template>
  <div
    ref="modal-backdrop"
    class="fixed-top z-10 inset-0 overflow-y-auto bg-black bg-opacity-50 h-100"
  >
    <div class="d-flex justify-content-center align-items-center h-100">
      <b-card text-variant="muted" class="shadow-sm border-0 w-50">
        <template #header>
          <div class="d-flex justify-content-between align-items-center">
            <h2 class="mb-0">{{ title }}</h2>

            <button
              data-testid="modal-close"
              type="button"
              class="bg-transparent border-0 text-muted"
              @click="close"
            >
              <i class="bi bi-x fa-2x"></i>
            </button>
          </div>
        </template>

        <slot :close="close" />
      </b-card>
    </div>
  </div>
</template>
