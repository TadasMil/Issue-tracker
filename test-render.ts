import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
