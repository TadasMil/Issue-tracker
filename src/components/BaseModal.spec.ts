import { describe, it, expect, vi } from 'vitest'
import { render, fireEvent } from '../../test-render'

import BaseModal from './BaseModal.vue'

describe('BaseModal', () => {
  it('should render configured modal component', () => {
    const closeFn = vi.fn()

    const { html } = render(BaseModal, {
      props: {
        title: 'Modal Title',
        close: closeFn,
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should emit close modal event', async () => {
    const closeFn = vi.fn()

    const { getByTestId } = render(BaseModal, {
      props: {
        title: 'Modal Title',
        close: closeFn,
      },
    })

    const closeButton = getByTestId('modal-close')

    await fireEvent.click(closeButton)

    expect(closeFn).toHaveBeenCalled()
  })
})
