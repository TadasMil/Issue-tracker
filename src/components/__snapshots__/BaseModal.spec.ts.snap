// Vitest Snapshot v1

exports[`BaseModal > should render configured modal component 1`] = `
"<div class=\\"fixed-top z-10 inset-0 overflow-y-auto bg-black bg-opacity-50 h-100\\">
  <div class=\\"d-flex justify-content-center align-items-center h-100\\">
    <div class=\\"card text-muted shadow-sm border-0 w-50\\">
      <!---->
      <div class=\\"card-header\\">
        <div class=\\"d-flex justify-content-between align-items-center\\">
          <h2 class=\\"mb-0\\">Modal Title</h2><button data-testid=\\"modal-close\\" type=\\"button\\" class=\\"bg-transparent border-0 text-muted\\"><i class=\\"bi bi-x fa-2x\\"></i></button>
        </div>
      </div>
      <div class=\\"card-body\\">
        <!---->
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>"
`;
